components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -292,13 +294,40 @@ private void updateBasicApplicationData(int applicationId, String applicationNam
         }
 
         // only if the application has been renamed
-        if (!applicationName.equalsIgnoreCase(storedAppName)) {
+        if (!applicationName.equals(storedAppName)) {
             // rename the role
             ApplicationMgtUtil.renameRole(storedAppName, applicationName);
             if (debugMode) {
                 log.debug("Renaming application role from " + storedAppName + " to "
                         + applicationName);
             }
+            PreparedStatement readPermissions = null;
+            ResultSet resultSet = null;
+            try {
+                readPermissions = connection.prepareStatement(ApplicationMgtDBQueries.LOAD_UM_PERMISSIONS);
+                readPermissions.setString(1, "%" + ApplicationMgtUtil.getApplicationPermissionPath() + "%");
+                resultSet = readPermissions.executeQuery();
+                while (resultSet.next()) {
```

Comment:
Break this logic to two separate methods, which would make the code more readable.